### ISSUE TITLE
Check for permissions of files before attempting to access them

### ISSUE THREAD

**Issue by wwood** [wurmlab/sequenceserver#104]
As reported at
https://groups.google.com/forum/#!msg/sequenceserver/Zd7fI_VfszQ/jFI-4xUFQMcJ

Seqserv should check for (read/execute but not write) permissions of 
- db folder and databases inside
- configuration file

**Comment by yeban** [comment]
Partly resolved in abaed275435a926822d53761b4744f5cfc142c64.

If SequenceServer couldn't read config file, it quits with the following message:
```
Error in config file: /Users/priyam/.sequenceserver.conf.
Permission denied @ rb_sysopen - /Users/priyam/.sequenceserver.conf
```

I could change the first line to read "Error reading config file: ..." which covers all cases (including bad indent in YAML). I have no control over the second line.

The rest will be resolved in #133, I think.

**Comment by yeban** [comment]
Instead of checking permissions, relevant exceptions (e.g., EACCESS) should be handled on access. This automatically happens as our code base matures over time. And checking for permissions at startup, or via `--doctor` doesn't solve the problem.
